airbyte-commons-worker/src/main/java/io/airbyte/workers/process/KubeProcessFactory.java

```diff
@@ -108,7 +108,11 @@ public Process create(
 
       final var allLabels = getLabels(jobId, attempt, customLabels);
 
-      final var nodeSelectors = usesIsolatedPool ? workerConfigs.getWorkerIsolatedKubeNodeSelectors() : workerConfigs.getworkerKubeNodeSelectors();
+      // If using isolated pool, check workerConfigs has isolated pool set. If not set, fall back to use
+      // regular node pool.
+      final var nodeSelectors =
+          usesIsolatedPool ? workerConfigs.getWorkerIsolatedKubeNodeSelectors().orElse(workerConfigs.getworkerKubeNodeSelectors())
```

Comment:
good call! fixed

Reply:
Realized this is not true - usesIsolatedPool here is actually passed by callers (isCustomConnector flag). Thus for OSS users who do not run them in a separate pool, usesIsolatedPool might still be true because they are custom connectors. I should rename this to isCustomConnectors instead.

Reply:
Oh good call, OSS users should be able to use custom connectors without an isolated pool indeed